### PR TITLE
Change note value #24

### PR DESCRIPTION
**Pull request by dtebbs**
- Constant defining the value of units in zeth notes
- Switch unit vlue from 1 Wei to 1 Szabo

Addresses issue #24

### CODE REVIEW

File: zeth-contracts/contracts/BaseMixer.sol [hunk 1]
```diff
@@ -64,6 +64,10 @@ contract BaseMixer is MerkleTreeMiMC7, ERC223ReceivingContract {
     // If token = address(0) then the mixer works with ether
     address public token;
 
+    // The unit used for public values (ether in and out), in Wei.  Must match
```
Comment (AntoineRondelet) [inline zeth-contracts/contracts/BaseMixer.sol]: Please fix the double spacing after each `.` in the files of this PR

Comment (dtebbs) [reply zeth-contracts/contracts/BaseMixer.sol]: Replaced.

File: zeth-contracts/contracts/BaseMixer.sol [hunk 2]
```diff
@@ -216,7 +222,9 @@ contract BaseMixer is MerkleTreeMiMC7, ERC223ReceivingContract {
         }
 
         // 2. Get vpub_out in wei
-        uint64 vpub_out = Bytes.get_value_from_inputs(Bytes.int256ToBytes8(primary_inputs[2*(1 + jsIn + jsOut)]));
+        uint64 vpub_out_raw = Bytes.get_value_from_inputs(
```
[the remark this answers is not in the document]
Comment (dtebbs) [reply zeth-contracts/contracts/BaseMixer.sol]: Renamed the variable

File: zeth-contracts/contracts/BaseMixer.sol [hunk 3]
```diff
@@ -64,6 +64,10 @@ contract BaseMixer is MerkleTreeMiMC7, ERC223ReceivingContract {
     // If token = address(0) then the mixer works with ether
     address public token;
 
+    // The unit used for public values (ether in and out), in Wei. Must match
+    // the python wrappers. Use Szabos (10^12 Wei).
+    uint64 constant public_unit_value_wei = 1000000000000;
```
[the remark this answers is not in the document]
Comment (dtebbs) [reply zeth-contracts/contracts/BaseMixer.sol]: Changed to `1 szabo`

[the remark this answers is not in the document]
Comment (AntoineRondelet) [reply zeth-contracts/contracts/BaseMixer.sol]: Great, thanks @dtebbs 
Could you write a test or modify the current ones to make sure the fix is functional. The goal is to do a payment of, say, more than 100ETH to make sure that we can now make payments of important denomination as flagged here: https://github.com/clearmatics/zeth/issues/24

[the remark this answers is not in the document]
Comment (dtebbs) [reply zeth-contracts/contracts/BaseMixer.sol]: Changed the values of the current tests to deal with values > 100

**Comment by AntoineRondelet** [comment]
Why do you have all the mpc code in here? Wrong rebase @dtebbs ?

**Comment by dtebbs** [comment]
> Why do you have all the mpc code in here? Wrong rebase @dtebbs ?

@AntoineRondelet  Ooops.  Yes, looks like a wrong rebase.  Sorry about that.  Should be fixed now.

**Comment by AntoineRondelet** [comment]
No problem, thanks, looks repaired now :)

File: pyClient/testERCTokenMixing.py [hunk 4]
```diff
@@ -132,12 +135,16 @@ def mint_token(token_instance, spender_address, deployer_address, token_amount):
...
+    tx_hash = approve(
+        token_instance,
+        bob_eth_address,
+        mixer_instance.address,
+        zethTest.BOB_DEPOSIT_ETH)
     w3.eth.waitForTransactionReceipt(tx_hash)
     allowance_mixer = allowance(token_instance, bob_eth_address, mixer_instance.address)
     print("- The allowance for the Mixer from Bob is:", allowance_mixer)
-    # Bob deposits 4ETH split in 2 notes of denominations of 2ETh and 2ETH on the mixer
+    # Bob deposits ETHToken, split in 2 notes of denominations on the mixer
```
Comment (AntoineRondelet) [inline pyClient/testERCTokenMixing.py]: `# Bob deposits ETHToken, split in 2 notes of denominations on the mixer` -> `# Bob deposits BOB_DEPOSIT_ETH, split in 2 notes of denominations BOB_SPLIT_1_ETH and BOB_SPLIT_2_ETH resp. on the mixer` ?

File: pyClient/testERCTokenMixing.py [hunk 5]
```diff
@@ -173,7 +180,7 @@ def mint_token(token_instance, spender_address, deployer_address, token_amount):
     assert (recovered_plaintext1 == ""),"Alice managed to decrypt a ciphertext that was not encrypted with her key!"
     assert (recovered_plaintext2 == ""),"Alice managed to decrypt a ciphertext that was not encrypted with her key!"
 
-    # Bob does a transfer of 1 ETHToken to Charlie on the mixer
+    # Bob does a transfer of ETHToken to Charlie on the mixer
```
Comment (AntoineRondelet) [inline pyClient/testERCTokenMixing.py]: `# Bob does a transfer of BOB_TO_CHARLIE_ETH to Charlie on the mixer`?

File: zeth-contracts/contracts/BaseMixer.sol (hunk 3 above)
[the remark this answers is not in the document]
Comment (AntoineRondelet) [reply zeth-contracts/contracts/BaseMixer.sol]: Thanks

**Comment by dtebbs** [comment]
> Not all comments have been updated in the tests

I think I have caught all the places now.  I checked the comments in `testEtherMixing` and `testERCTOkenMixing`, and opted to remove the values to save the maintenance burden of keeping them consistent.

**Comment by AntoineRondelet** [comment]
> > Not all comments have been updated in the tests
> 
> I think I have caught all the places now. I checked the comments in `testEtherMixing` and `testERCTOkenMixing`, and opted to remove the values to save the maintenance burden of keeping them consistent.

Yes makes sense. The amounts should already be specified in the `prints` of each test case anyway, so there shouldn't be a need to re-specify them in the comments.

**Comment by AntoineRondelet** [comment]
LGTM

**Pull request merged**